spring-web/src/main/java/org/springframework/web/context/support/AnnotationConfigGenericWebApplicationContext.java

```diff
@@ -0,0 +1,168 @@
+package org.springframework.web.context.support;
+
+import java.util.Arrays;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.function.Supplier;
+
+import javax.annotation.Nonnull;
+
+import org.springframework.beans.factory.config.BeanDefinitionCustomizer;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.beans.factory.support.BeanNameGenerator;
+import org.springframework.beans.factory.support.DefaultListableBeanFactory;
+import org.springframework.boot.web.servlet.context.AnnotationConfigServletWebServerApplicationContext;
+import org.springframework.context.annotation.AnnotatedBeanDefinitionReader;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.context.annotation.AnnotationConfigRegistry;
+import org.springframework.context.annotation.AnnotationConfigUtils;
+import org.springframework.context.annotation.AnnotationScopeMetadataResolver;
+import org.springframework.context.annotation.ClassPathBeanDefinitionScanner;
+import org.springframework.context.annotation.ScopeMetadataResolver;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.util.Assert;
+import org.springframework.util.ClassUtils;
+import org.springframework.web.context.support.GenericWebApplicationContext;
+
+/**
+ * A copy of Spring-boot 2.1.8 {@link AnnotationConfigServletWebServerApplicationContext},
+ * except that it does not use its own WebServer (other super class).
+ * <p>
+ * Also See {@link AnnotationConfigApplicationContext} (Same as this but uses
+ * GenericApplicationContext instead of GenericWebApplicationContext)
+ * <p>
+ * In contrast to {@link org.springframework.web.context.support.AnnotationConfigWebApplicationContext},
+ * this class uses the GenericWebApplicationContext, which allows registering beans using a Supplier
+ * {@link #registerBean(Class, Supplier, BeanDefinitionCustomizer...)}
+ * <p>
+ * See javadoc of superclass {@link GenericWebApplicationContext} for more information.
```

Comment:
* {@link GenericWebApplicationContext} that accepts annotated classes as input - in
 * particular {@link org.springframework.context.annotation.Configuration @Configuration}
 * -annotated classes, but also plain {@link Component @Component} classes and JSR-330
 * compliant classes using {@code javax.inject} annotations. Allows for registering
 * classes one by one (specifying class names as config location) as well as for classpath
 * scanning (specifying base packages as config location).
 * <p>
 * See @{@link Configuration}'s javadoc for usage examples.
 * <p>
 * Note: In case of multiple {@code @Configuration} classes, later {@code @Bean}
 * definitions will override ones defined in earlier loaded files. This can be leveraged
 * to deliberately override certain bean definitions via an extra Configuration class.